pkg/pgmodel/ingestor/watcher.go

```diff
@@ -0,0 +1,107 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package ingestor
+
+import (
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/timescale/promscale/pkg/log"
+)
+
+const (
+	// reportThreshold is a value above which the warn would be logged. This value is
+	// the value of the ratio between incoming batches vs outgoing batches.
+	reportRatioThreshold = 3
+	refreshIteration     = 10 // This means that values of watcher.incomingBatches and watcher.outgoingBatches gets renewed every (refreshIteration * checkThroughputInterval).
+	checkRatioInterval   = time.Minute
+
+	// Duration warnings constants.
+	reportDurationThreshold        = time.Minute
+	checkIngestionDurationInterval = time.Minute
+)
+
+// throughtputWatcher is a light weight samples batch watching type, that serves as a watching
+// routine and keeps the track of incoming write batches. It keeps a ratio of
+// incoming samples vs outgoing samples and warn about low throughput to the user
+// which might be due to external causes like network latency, resources allocated, etc.
+//
+// We watch the batches than the samples in total, since watching batching is fast
+// and does the same purpose of watching samples, but on a higher-level.
+type throughtputWatcher struct {
+	// Warn based on ratio.
+	incomingBatches uint64
```

Comment:
should we use https://github.com/prometheus/prometheus/blob/main/storage/remote/ewma.go here instead?

Reply:
It seems like we want to compare smoothed rates. instead of total batches.

Reply:
Sorry but I don't understand what's wrong with the current approach? The current way is simple and easier to maintain.

Reply:
TODO